### PR TITLE
feat(dashpay): add timeskew support to CoinJoinService and warning dialogs

### PR DESCRIPTION
**Pull request by HashEngineering**
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -532,13 +547,6 @@ class MainActivity : AbstractBindServiceActivity(), ActivityCompat.OnRequestPerm
         }
     }
 
-    private fun checkLowStorageAlert() {
-        val stickyIntent = registerReceiver(null, IntentFilter(Intent.ACTION_DEVICE_STORAGE_LOW))
-        if (stickyIntent != null) {
-            showLowStorageAlertDialog()
-        }
-    }
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/main/MainActivity.kt:540]: DashPay was using the old checkLowStorageAlert()

File: wallet/src/de/schildbach/wallet/ui/main/MainViewModel.kt [hunk 2]
```diff
@@ -379,16 +381,26 @@ class MainViewModel @Inject constructor(
...
             val networkTime = result.headers.getDate("date")?.time
             requireNotNull(networkTime)
-            abs(systemTimeMillis - networkTime)
+            val maxAllowedTimeSkew = if (coinJoinConfig.getMode() == CoinJoinMode.NONE) {
+                TIME_SKEW_TOLERANCE
+            } else {
+                TIME_SKEW_TOLERANCE_COINJOIN
+            }
+            val timeSkew = abs(systemTimeMillis - networkTime) / 1000
+            return Pair(timeSkew > maxAllowedTimeSkew, timeSkew)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/main/MainViewModel.kt:400]: for the MainActivity we will show one message if CoinJoin is ON, and another if OFF.  Each has different tolerances.

File: wallet/src/de/schildbach/wallet/service/CoinJoinService.kt [hunk 3]
```diff
@@ -235,26 +262,28 @@ class CoinJoinMixingService @Inject constructor(
...
+                "coinjoin-new-state: $mode, ${timeSkew / 1000}s, $hasAnonymizableBalance, $networkStatus, synced: $isSynced, ${blockChain != null}"
             )
             this.networkStatus = networkStatus
             this.hasAnonymizableBalance = hasAnonymizableBalance
             this.isSynced = isSynced
             this.mode = mode
+            this.timeSkew = timeSkew
 
-            if (mode == CoinJoinMode.NONE) {
+            if (mode == CoinJoinMode.NONE || timeSkew > MAX_ALLOWED_TIMESKEW) {
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/service/CoinJoinService.kt:286]: we will use `timeSkew` to determine if CoinJoin will actually run or not.

`timeSkew` > 2 seconds will give the same result as CoinJoin OFF.

**Review by Syn-McJ — APPROVED** [review]
Looks good

**Pull request merged**